p2p/simulations/http.go

```diff
@@ -410,9 +412,38 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// check if filtering has been requested
+	if fp := req.URL.Query().Get("filterProtocol"); fp != "" {
+		if fc := req.URL.Query().Get("filterCode"); fc != "" {
+			strs := strings.Split(fc, ",")
+			uints := make([]uint64, len(strs))
+			for i := range uints {
+				u, err := strconv.ParseUint(strs[i], 10, 64)
+				if err != nil {
+					http.Error(w, "Invalid msg code for filtering provided", http.StatusBadRequest)
+					return
+				}
+				uints = append(uints, u)
+			}
+			activeMsgFilters[fp] = uints
+		} else {
```

Comment:
It should be explicit (so not specifying it is an error), and perhaps accept `*` or `-1` to mean "give me all messages regardless of code".